rpcs3/Emu/RSX/GL/GLPresent.cpp

```diff
@@ -330,19 +330,32 @@ void GLGSRender::flip(const rsx::display_flip_info_t& info)
 			const f32 gamma = avconfig.gamma;
 			const bool limited_range = !use_full_rgb_range_output;
 			const auto filter = m_output_scaling == output_scaling_mode::nearest ? gl::filter::nearest : gl::filter::linear;
-			rsx::simple_array<gl::texture*> images{ image_to_flip, image_to_flip2 };
+			rsx::simple_array<gl::texture*> images{ image_to_flip };
+
+			if (image_to_flip2)
+			{
+				images.push_back(image_to_flip2);
+			}
 
 			if (m_output_scaling == output_scaling_mode::fsr && avconfig.stereo_mode == stereo_render_mode_options::disabled) // 3D will be implemented later
 			{
-				for (unsigned i = 0; i < 2 && images[i]; ++i)
+				for (unsigned i = 0; i < images.size() && images[i]; ++i)
 				{
 					const rsx::flags32_t mode = (i == 0) ? UPSCALE_LEFT_VIEW : UPSCALE_RIGHT_VIEW;
 					images[i] = m_upscaler->scale_output(cmd, image_to_flip, screen_area, aspect_ratio.flipped_vertical(), mode);
 				}
 			}
 
 			gl::screen.bind();
-			m_video_output_pass.run(cmd, areau(aspect_ratio), images.map(FN(x->id())), gamma, limited_range, avconfig.stereo_mode, filter);
```

Comment:
The problem is obviously with the map callback. Why not just change it to `FN(x ? x->id() : GL_NONE)`? Having a nullptr in a list of pointers shouldn't break things.

Reply:
ah... it's a nonsensical shortcut macro for a simple lambda... my IDE was pointing me to opengl code for some reason